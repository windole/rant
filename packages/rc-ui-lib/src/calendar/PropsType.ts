import React, { MouseEvent } from 'react';
import { PopupPosition } from '../popup';
import { BaseTypeProps, TeleportType } from '../utils';

export type CalendarType = 'single' | 'range' | 'multiple';

export type CalendarDayType =
  | ''
  | 'start'
  | 'start-end'
  | 'middle'
  | 'end'
  | 'selected'
  | 'multiple-middle'
  | 'multiple-selected'
  | 'disabled'
  | 'placeholder';

export type CalendarDayItem = {
  date?: Date;
  text?: string | number;
  type?: CalendarDayType;
  topInfo?: string;
  className?: string;
  bottomInfo?: string;
};

export type CalendarDayProps = {
  item: CalendarDayItem;
  color: string;
  index: number;
  offset?: number;
  rowHeight: string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarInstance = {
  reset: (date?: Date | Date[]) => void;
  scrollToDate: (targetDate: Date) => void;
};

export interface CalendarProps extends BaseTypeProps {
  position?: PopupPosition;
  round?: boolean;
  closeOnPopstate?: boolean;
  closeOnClickOverlay?: boolean;
  safeAreaInsetBottom?: boolean;
  teleport?: TeleportType;
  show?: boolean;
  type?: CalendarType;
  title?: React.ReactNode | string;
  subtitle?: React.ReactNode | string;
  color?: string;
  readonly?: boolean;
  poppable?: boolean;
  maxRange?: string | number;
  showMark?: boolean;
  showTitle?: boolean;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  rowHeight?: string | number;
  confirmText?: string;
  rangePrompt?: string;
  lazyRender?: boolean;
  showConfirm?: boolean;
  defaultDate?: Date | Date[] | null;
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showRangePrompt?: boolean;
  confirmDisabledText?: string;
  minDate?: Date;
  maxDate?: Date;
  firstDayOfWeek?: number;
  footer?: React.ReactNode | string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClickSubtitle?: (ev: MouseEvent) => void;
  /** 点击确认后的回调 */
  onConfirm?: (v: Date | Date[]) => void;
  onOverRange?: () => void;
  onSelect?: (v: Date | Date[]) => void;
  onUnselect?: (v: Date | Date[]) => void;
  onMonthShow?: (obj: { date: Date; title: string }) => void;
  onClose?: () => void;
}

export type CalendarMonthProps = {
  date?: Date;
  type?: CalendarType;
  color?: string;
  minDate?: Date;
  maxDate?: Date;
  showMark?: boolean;
  rowHeight?: number | string;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  formatMonthTitle?: (date: Date) => React.ReactNode;
  lazyRender?: boolean;
  currentDate?: Date | Date[];
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showMonthTitle?: boolean;
  firstDayOfWeek?: number;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarMonthInstance = {
  showed?: boolean;
  getTitle: () => string | React.ReactNode;
  getHeight: () => number;
  setVisible: (value?: boolean | undefined) => void;
  scrollToDate: (body: Element, targetDate: Date) => void;
};
