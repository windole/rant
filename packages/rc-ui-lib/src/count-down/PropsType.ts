import React from 'react';
import { CurrentTime } from '../hooks/use-count-down';
import { BaseTypeProps } from '../utils';

export interface CountDownProps extends BaseTypeProps {
  /** 	倒计时时长，单位毫秒 */
  time?: number | string;
  /** 时间格式	 */
  format?: string;
  /**  是否自动开始倒计时 */
  autoStart?: boolean;
  /**  是否开启毫秒级渲染 */
  millisecond?: boolean;
  /**  倒计时结束时触发 */
  onFinish?: () => void;
  /**  是否开启毫秒级渲染 */
  onChange?: (value: CurrentTime) => void;
  renderChildren?: (value: CurrentTime) => React.ReactNode;
}

export type CountDownCurrentTime = CurrentTime;

export type CountDownInstance = {
  start: () => void;
  pause: () => void;
  reset: () => void;
};
