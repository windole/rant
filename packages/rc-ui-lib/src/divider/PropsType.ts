import React from 'react';
import { BaseTypeProps } from '../utils';

export type DividerContentPosition = 'left' | 'center' | 'right';
export interface DividerProps extends BaseTypeProps, React.HTMLAttributes<HTMLDivElement> {
  /** 是否使用虚线	 */
  dashed?: boolean;
  /**
   * 是否使用 0.5px 线
   * @default true
   */
  hairline?: boolean;
  /**
   * 内容位置
   * @default 'center'
   */
  contentPosition?: DividerContentPosition;
}
