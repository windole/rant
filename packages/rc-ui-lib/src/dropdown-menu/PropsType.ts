import React, { ReactNode } from 'react';
import { BaseTypeProps, TeleportType } from '../utils';

type DirectionTypes = 'up' | 'down';

export type DropdownItemOption = {
  /** 文案 */
  text?: string;
  /** 标识符 */
  value?: string | number;
  /** 左侧图标名称或图片链接 */
  icon?: string | ReactNode;
};

export interface DropdownMenuProps extends BaseTypeProps {
  /** 菜单标题和选项的选中态颜色 */
  activeColor?: string;
  /** 菜单展开方向 */
  direction?: DirectionTypes;
  /** 菜单栏 z-index 层级 */
  zIndex?: string | number;
  /** 动画时长，单位秒 */
  duration?: number | string;
  /** 是否显示遮罩层 */
  overlay?: boolean;
  /** 是否在点击遮罩层后关闭菜单 */
  closeOnClickOverlay?: boolean;
  /** 是否在点击外部元素后关闭菜单 */
  closeOnClickOutside?: boolean;
  /** 子元素 */
  children?: React.ReactNode;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (value: DropdownItemOption) => void;
  /** @private */
  shouldRenderTitle?: boolean;
}

export interface DropdownItemProps extends BaseTypeProps {
  /** 标志符 */
  name?: string;
  /** 菜单项标题 */
  title?: string;
  /** 默认值 */
  defaultValue?: DropdownItemOption;
  /** 当前选中项对应的 value */
  value?: DropdownItemOption;
  /** 占位文本 */
  placeholder?: string;
  /** 选项数组 */
  options?: DropdownItemOption[];
  /** 是否禁用菜单 */
  disabled?: boolean;
  /** 标题额外类名 */
  titleClass?: string;
  /** 子元素 */
  children?: React.ReactNode;
  /** @private */
  showPopup?: boolean;
  /** 指定挂载的节点 */
  teleport?: TeleportType;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (value: DropdownItemOption) => void;
  /** 打开菜单栏时触发 */
  onOpen?: () => void;
  /** 打开菜单栏且动画结束后触发 */
  onOpened?: () => void;
  /** 关闭菜单栏时触发 */
  onClose?: () => void;
  /** 关闭菜单栏且动画结束后触发 */
  onClosed?: () => void;
}

export interface DropdownMenuContextProps {
  props?: React.PropsWithChildren<DropdownMenuProps>;
  value?: Record<string, number | string>;
  offset?: number | string;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (obj) => void;
  renderTitle?: (ins: DropdownItemInstance) => void;
  close?: () => void;
}

export interface DropdownMenuInstance {
  close: () => void;
}
export interface DropdownItemInstance {
  /** 切换菜单展示状态，传 true 为显示，false 为隐藏，不传参为取反	 */
  toggle: (
    show?: boolean,
    options?: {
      immediate?: boolean;
    },
  ) => void;
  /** @private */
  state: {
    showPopup: boolean;
    transition: boolean;
    showWrapper: boolean;
  };
  /** @private */
  renderTitle: (value) => string | ReactNode;
}
