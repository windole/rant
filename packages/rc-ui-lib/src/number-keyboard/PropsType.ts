import React from 'react';
import { TeleportType } from '../utils';

export type NumberKeyboardTheme = 'default' | 'custom';

export type KeyType = '' | 'delete' | 'extra' | 'close';

export type KeyConfig = {
  text?: number | string;
  type?: KeyType;
  color?: string;
  wider?: boolean;
};

export interface NumberKeyboardProps {
  /** 是否显示键盘 */
  visible?: boolean;
  /** 键盘标题 */
  title?: string;
  /** 样式风格，可选值为 `custom`、`default` */
  theme?: NumberKeyboardTheme;
  /** 键盘 z-index 层级 */
  zIndex?: number;
  /**
   * 是否在点击关闭按钮时触发 blur 事件
   * 默认值为`true`
   */
  blurOnClose?: boolean;
  /**
   * 是否展示删除图标
   * 默认值为`true`
   */
  showDeleteKey?: boolean;
  /** 是否将通过随机顺序展示按键 */
  randomKeyOrder?: boolean;
  /** 关闭按钮文字，空则不展示 */
  closeButtonText?: string;
  /** 删除按钮文字，空则展示删除图标 */
  deleteButtonText?: string;
  /**
   * 是否将关闭按钮设置为加载中状态，
   * 仅在 `theme="custom"` 时有效
   */
  closeButtonLoading?: boolean;
  /**
   * 是否在点击外部时收起键盘
   * 默认值为`true`
   */
  hideOnClickOutside?: boolean;
  /**
   * 是否开启底部安全区适配
   * 默认值为`true`
   */
  safeAreaInsetBottom?: boolean;
  /** 底部额外按键的内容 */
  extraKey?: string | string[];
  /**
   * 自定义标题左侧
   */
  titleLeft?: React.ReactNode;
  /**
   * 点击关闭按钮时触发
   */
  onClose?: () => void;
  /** 点击按键时触发 */
  onInput?: (v: string) => void;
  /** 点击删除键时触发 */
  onDelete?: () => void;
  /** 点击关闭按钮或非键盘区域时触发 */
  onBlur?: () => void;
  /** 指定挂载的节点 */
  teleport?: TeleportType;
}
