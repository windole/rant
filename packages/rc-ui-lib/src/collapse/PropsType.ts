import { ReactNode } from 'react';
import { CellProps } from '../cell/PropsType';
import { BaseTypeProps } from '../utils';

export interface CollapseProps extends BaseTypeProps {
  /** 是否开启手风琴模式	 */
  accordion?: boolean;
  /** 默认打开的面板name */
  initValue?: string | number | Array<number | string>;
  /** 是否显示外边框	 */
  border?: boolean;
  /** 当前展开面板的 name	 */
  value?: number | string | Array<number | string>;
  /** 子组件	 */
  children?: ReactNode;
  /** 切换面板时触发	*/
  onChange?: (activeNames: number | string | Array<number | string>) => void;
}

export interface CollapseItemProps extends CellProps {
  /** 唯一标识符，默认为索引值	 */
  name?: number | string;
  /** 是否禁用面板	 */
  disabled?: boolean;
  /** 是否为只读状态，只读状态下无法操作面板	 */
  readonly?: boolean;
  /** @private */
  index?: number;
  /** 子组件	 */
  children?: ReactNode;
  // onClick: () => void;
}

export type CollapseItemInstance = {
  toggle: (expand: boolean) => void;
};
