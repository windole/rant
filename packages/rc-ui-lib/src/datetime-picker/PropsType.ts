import { PickerInstance, PickerProps } from '../picker/PropsType';

export type DatetimePickerColumnType = 'year' | 'month' | 'day' | 'hour' | 'minute';

export type DatetimePickerType =
  | 'date'
  | 'time'
  | 'datetime'
  | 'datehour'
  | 'month-day'
  | 'year-month';

export interface SharedProps extends Omit<PickerProps, 'onConfirm' | 'onChange' | 'columns'> {
  value?: string | Date;
  /** 选项过滤函数	 */
  filter?: (type: string, values: string[]) => string[];
  /**
   * 自定义列排序数组,
   * 子项可选值为: year、month、day、hour、minute
   */
  columnsOrder?: string[];
  /** 选项格式化函数	*/
  formatter?: (type: string, value: string) => string;
  /** 点击完成按钮时触发的事件	 */
  onConfirm?: (value: Date) => void;
  /** 当值变化时触发的事件	 */
  onChange?: (value: Date) => void;
  onCancel?: () => void;
}

export interface DatePickerProps extends SharedProps {
  /** 时间类型 */
  type?: DatetimePickerType;
  /** 可选的最小时间，精确到分钟	 */
  minDate?: Date;
  /** 可选的最大时间，精确到分钟	 */
  maxDate?: Date;
}

export interface TimePickerProps extends SharedProps {
  /** 可选的最小小时	 */
  minHour?: number | string;
  /** 可选的最大小时	 */
  maxHour?: number | string;
  /** 可选的最小分钟	 */
  minMinute?: number | string;
  /** 可选的最大分钟	 */
  maxMinute?: number | string;
}

export type DateTimePickerProps = DatePickerProps & TimePickerProps;

export type DateTimePickerInstance = {
  getPicker: () => PickerInstance;
};
