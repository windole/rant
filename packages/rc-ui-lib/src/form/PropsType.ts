import React from 'react';
import type { FormProps as RcFormProps, FormInstance as RcFormInstance } from 'rc-field-form';
import type { FieldProps as RcFieldProps } from 'rc-field-form/lib/Field';
import type { Meta } from 'rc-field-form/lib/interface';
import type { BaseTypeProps } from '../utils';
import type { CellGroupProps } from '../cell/PropsType';
import { FieldProps } from '../field/PropsType';

export type FormLayout = 'vertical' | 'horizontal';

export type FormValidateTrigger = 'onBlur' | 'onChange' | 'onSubmit';
type OmitFromCellGroupProps = Omit<CellGroupProps, 'title'>;
export interface FormProps extends RcFormProps, OmitFromCellGroupProps, BaseTypeProps {
  /** 表单布局 */
  layout?: FormLayout;
  /** 是否显示 label 后面的冒号 */
  colon?: boolean;
  /** 表单底部内容	 */
  footer?: React.ReactNode;
  /** 是否显示验证信息 */
  showValidateMessage?: boolean;
  /** 子元素 */
  children?: React.ReactNode;
  validateTrigger?: FormValidateTrigger;
}

export type RenderChildren<Values = unknown> = (form: RcFormInstance<Values>) => React.ReactNode;
type ChildrenType<Values = unknown> = RenderChildren<Values> | React.ReactNode;

export type FormInstance = RcFormInstance;

export type MemoInputProps = {
  value: unknown;
  update: number;
  children: React.ReactNode;
} & Record<string, unknown>;

export type FormItemProps = RcFieldProps &
  Pick<FormProps, 'showValidateMessage' | 'validateTrigger'> &
  Pick<
    FieldProps,
    | 'style'
    | 'className'
    | 'tooltip'
    | 'intro'
    | 'colon'
    | 'labelWidth'
    | 'labelAlign'
    | 'labelClass'
    | 'onClick'
  > & {
    /** 输入框左侧文本 */
    label?: string;
    /** 必填项 */
    required?: boolean;
    noStyle?: boolean;
    /** 是否禁用表单项 */
    disabled?: boolean;
    /** 自定义item，此时不会渲染内置的field */
    customField?: boolean;
    /** 子元素 */
    children?: ChildrenType;
  };

export type FormItemLayoutProps = Pick<
  FormItemProps,
  | 'required'
  | 'disabled'
  | 'label'
  | 'style'
  | 'className'
  | 'tooltip'
  | 'intro'
  | 'colon'
  | 'labelWidth'
  | 'labelAlign'
  | 'labelClass'
  | 'showValidateMessage'
  | 'validateTrigger'
> & {
  onClick?: (e?: React.MouseEvent) => void;
  htmlFor?: string;
  meta?: Meta;
  layout?: FormLayout;
  /** @private */
  isFieldChildren?: boolean;
};
