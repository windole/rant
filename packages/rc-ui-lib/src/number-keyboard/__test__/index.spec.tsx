import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { NumberKeyboard, NumberKeyboardProps } from '..';
import { NumberKeyboardTheme } from '../PropsType';
import TestsEvent from '../../../tests/events';

const clickKey = async (target) => {
  await TestsEvent.triggerTouch(target, 'touchstart', [[0, 0]]);
  await TestsEvent.triggerTouch(target, 'touchend', [[0, 0]]);
};

const $props = {
  visible: true,
};

describe('NumberKeyboard', () => {
  function createNumberKeyboard(props: NumberKeyboardProps) {
    const { baseElement, getAllByRole, getByText, container, rerender, debug } = render(
      <NumberKeyboard {...props} />,
    );

    return {
      baseElement,
      container,
      rerender,
      getByText,
      getAllByRole,
      debug,
    };
  }

  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should emit input event after clicking number key', async () => {
    const onInput = jest.fn();
    const props = {
      ...$props,
      onInput,
    };

    const { getAllByRole } = createNumberKeyboard(props);
    const key = getAllByRole('button')[0];
    await clickKey(key);
    expect(onInput).toHaveBeenCalledWith('1');
  });

  it('should emit onDelete event after clicking delete key', async () => {
    const onDelete = jest.fn();
    const props = {
      ...$props,
      onDelete,
    };

    const { getAllByRole } = createNumberKeyboard(props);
    const key = getAllByRole('button')[11];
    await clickKey(key);
    expect(onDelete).toHaveBeenCalled();
  });

  it('should emit onBlur event after clicking collapse key', async () => {
    const onBlur = jest.fn();
    const props = {
      ...$props,
      onBlur,
    };

    const { getAllByRole } = createNumberKeyboard(props);
    const key = getAllByRole('button')[9];
    await clickKey(key);
    expect(onBlur).toHaveBeenCalled();
  });

  it('should emit close event after clicking close button', async () => {
    const onBlur = jest.fn();
    const onClose = jest.fn();
    const props = {
      ...$props,
      theme: 'custom' as NumberKeyboardTheme,
      onBlur,
      onClose,
    };

    const { getAllByRole } = createNumberKeyboard(props);
    const key = getAllByRole('button')[12];
    await clickKey(key);
    expect(onBlur).toHaveBeenCalled();
    expect(onClose).toHaveBeenCalled();
  });

  it('should render title and close button correctly', () => {
    const props = {
      ...$props,
      title: 'Title',
      closeButtonText: 'Close',
    };

    const { baseElement } = createNumberKeyboard(props);
    expect(baseElement.querySelector('.rc-number-keyboard__header')).toMatchSnapshot();
  });

  it('should render titleLeft prop correctly', () => {
    const props = {
      ...$props,
      titleLeft: <span>Custom Title Left</span>,
    };

    const { baseElement } = createNumberKeyboard(props);
    expect(baseElement.querySelector('.rc-number-keyboard__header')).toMatchSnapshot();
  });

  it('should render loading icon when using close-button-loading prop', () => {
    const props = {
      ...$props,
      theme: 'custom' as NumberKeyboardTheme,
      closeButtonLoading: true,
    };

    const { baseElement } = createNumberKeyboard(props);

    expect(baseElement.querySelector('.rc-key__loading-icon')).toBeTruthy();
  });

  it('should render extra key correctly when using extraKey prop', () => {
    const props = {
      ...$props,
      extraKey: 'foo',
    };

    const { baseElement } = createNumberKeyboard(props);
    expect(baseElement.querySelectorAll('.rc-key')[9]).toMatchSnapshot();
  });

  it('should render extra key correctly when using extraKey prop and extraKey is an array', async () => {
    const props = {
      ...$props,
      theme: 'custom' as NumberKeyboardTheme,
      extraKey: ['00', '.'],
    };

    const { baseElement } = createNumberKeyboard(props);

    expect(baseElement.querySelector('.rc-number-keyboard')).toMatchSnapshot();
  });

  it('should emit blur event after clicking outside', async () => {
    const onBlur = jest.fn();
    const props = {
      ...$props,
      onBlur,
    };

    createNumberKeyboard(props);
    await clickKey(document.body);
    expect(onBlur).toHaveBeenCalled();
  });

  it('should not emit blur event after clicking outside when hideOnClickOutside is false', async () => {
    const onBlur = jest.fn();
    const props = {
      ...$props,
      hideOnClickOutside: false,
      onBlur,
    };

    createNumberKeyboard(props);
    await clickKey(document.body);
    expect(onBlur).not.toHaveBeenCalled();
  });

  it('should add "rc-key--active" className to key when focused', async () => {
    const { getAllByRole } = createNumberKeyboard($props);
    const key = getAllByRole('button')[0];

    await TestsEvent.triggerTouch(key, 'touchstart', [[0, 0]]);

    expect(key.classList.contains('rc-key--active')).toBeTruthy();

    await TestsEvent.triggerTouch(key, 'touchend', [[0, 0]]);

    expect(key.classList.contains('rc-key--active')).toBeFalsy();
  });

  it('should remove "rc-key--active" className from key when touch moved', async () => {
    const { getAllByRole } = createNumberKeyboard($props);
    const key = getAllByRole('button')[0];

    await TestsEvent.triggerTouch(key, 'touchstart', [[0, 0]]);

    expect(key.classList.contains('rc-key--active')).toBeTruthy();

    await TestsEvent.triggerTouch(key, 'touchmove', [[0, 0]]);

    expect(key.classList.contains('rc-key--active')).toBeTruthy();

    await TestsEvent.triggerTouch(key, 'touchmove', [[100, 0]]);

    expect(key.classList.contains('rc-key--active')).toBeFalsy();

    await TestsEvent.triggerTouch(key, 'touchend', [[0, 0]]);

    expect(key.classList.contains('rc-key--active')).toBeFalsy();
  });

  it('should not render delete key when show-delete-key prop is false', () => {
    const props = {
      ...$props,
      showDeleteKey: true,
    };

    const { container } = createNumberKeyboard(props);

    expect(container.querySelector('.rc-key--delete')).toBeFalsy();
  });

  test('should shuffle key order when using random-key-order prop', () => {
    const props = {
      ...$props,
      randomKeyOrder: true,
    };

    const { getAllByRole } = createNumberKeyboard(props);

    const keys: number[] = [];
    const clickKeys: number[] = [];

    for (let i = 0; i < 9; i++) {
      keys.push(i + 1);
      clickKeys.push(Number(getAllByRole('button')[i].innerText));
    }

    expect(keys.every((v, k) => keys[k] === clickKeys[k])).toEqual(false);
  });
});
