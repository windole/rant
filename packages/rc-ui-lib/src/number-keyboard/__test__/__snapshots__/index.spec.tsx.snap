// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberKeyboard should render extra key correctly when using extraKey prop 1`] = `
<div
  class="rc-key"
  role="button"
  tabindex="0"
>
  foo
</div>
`;

exports[`NumberKeyboard should render extra key correctly when using extraKey prop and extraKey is an array 1`] = `
<div
  class="rc-number-keyboard"
>
  <div
    class="rc-number-keyboard__body"
  >
    <div
      class="rc-number-keyboard__keys"
    >
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          1
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          2
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          3
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          4
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          5
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          6
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          7
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          8
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          9
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          00
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          0
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key"
          role="button"
          tabindex="0"
        >
          .
        </div>
      </div>
    </div>
    <div
      class="rc-number-keyboard__sidebar"
    >
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key rc-key--large rc-key--delete"
          role="button"
          tabindex="0"
        >
          <svg
            class="rc-key__delete-icon"
            viewBox="0 0 32 22"
          >
            <path
              d="M28 0a4 4 0 0 1 4 4v14a4 4 0 0 1-4 4H10.4a2 2 0 0 1-1.4-.6L1 13.1c-.6-.5-.9-1.3-.9-2 0-1 .3-1.7.9-2.2L9 .6a2 2 0 0 1 1.4-.6zm0 2H10.4l-8.2 8.3a1 1 0 0 0-.3.7c0 .3.1.5.3.7l8.2 8.4H28a2 2 0 0 0 2-2V4c0-1.1-.9-2-2-2zm-5 4a1 1 0 0 1 .7.3 1 1 0 0 1 0 1.4L20.4 11l3.3 3.3c.2.2.3.5.3.7 0 .3-.1.5-.3.7a1 1 0 0 1-.7.3 1 1 0 0 1-.7-.3L19 12.4l-3.4 3.3a1 1 0 0 1-.6.3 1 1 0 0 1-.7-.3 1 1 0 0 1-.3-.7c0-.2.1-.5.3-.7l3.3-3.3-3.3-3.3A1 1 0 0 1 14 7c0-.3.1-.5.3-.7A1 1 0 0 1 15 6a1 1 0 0 1 .6.3L19 9.6l3.3-3.3A1 1 0 0 1 23 6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div
        class="rc-key__wrapper"
      >
        <div
          class="rc-key rc-key--blue rc-key--large"
          role="button"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberKeyboard should render title and close button correctly 1`] = `
<div
  class="rc-number-keyboard__header"
>
  <h2
    class="rc-number-keyboard__title"
  >
    Title
  </h2>
  <button
    class="rc-number-keyboard__close"
    type="button"
  >
    Close
  </button>
</div>
`;

exports[`NumberKeyboard should render titleLeft prop correctly 1`] = `
<div
  class="rc-number-keyboard__header"
>
  <span
    class="rc-number-keyboard__title-left"
  >
    <span>
      Custom Title Left
    </span>
  </span>
</div>
`;
